client/my-sites/upgrades/components/domain-warnings/index.jsx

```diff
@@ -65,18 +65,29 @@ export default React.createClass( {
 	},
 
 	getDomains() {
-		return ( this.props.domains || [ this.props.domain ] ).filter( domain => domain.isCurrentUserOwner );
+		return ( this.props.domains || [ this.props.domain ] );
 	},
 
 	expiredDomains() {
-		debug( 'Rendering expiredDomain' );
 		const expiredDomains = this.getDomains().filter( domain => domain.expired && domain.type === domainTypes.REGISTERED );
-		let text, renewLink;
+
 		if ( expiredDomains.length === 0 ) {
 			return null;
-		} else if ( expiredDomains.length === 1 ) {
+		}
+
+		if ( expiredDomains.some( domain => domain.isCurrentUserOwner ) ) {
+			return this.expiredDomainsOwner( expiredDomains.filter( domain => domain.isCurrentUserOwner ) );
+		}
+
+		return this.expiredDomainsNonOwner( expiredDomains.filter( domain => ! domain.isCurrentUserOwner ) );
+	},
+
+	expiredDomainsOwner( expiredDomains ) {
```

Comment:
I like the new var name in https://github.com/Automattic/wp-calypso/pull/6983 so it'll probably the latter.